modules/gdscript/language_server/gdscript_language_protocol.cpp

```diff
@@ -184,7 +184,9 @@ Dictionary GDScriptLanguageProtocol::initialize(const Dictionary &p_params) {
 	if (root_uri.length() && is_same_workspace) {
 		workspace->root_uri = root_uri;
 	} else {
-		workspace->root_uri = "file://" + workspace->root;
+		String r_root = workspace->root;
+		r_root = r_root.lstrip("/");
+		workspace->root_uri = "file:///" + r_root;
```

Comment:
Is this change still needed at all? This looks wrong and might break on Windows (and is unnecessary on Linux).

The be clear, the `file://` protocol is `file://` - two slashes. After that should come a valid URI.

On Linux a valid URI for an absolute path will start with `/` so it will indeed be `file:///home/akien/...` etc.
On Windows AFAIU a valid URI will start with `c:\\path\\to\\root` so it would become `file://c:\\path\\to\\root` which should be fine.

Reply:
@akien-mga If I understand @A-Lamia correctly, the problem is that this `root_uri` is later matched against other paths coming from the LSP client, and apparently they are sent like this on Windows clients: `file:///c:/path/to/root`.

Reply:
If that is the case, I wonder if this wouldn't be the less error prone way to do this?
```C
#ifndef WINDOWS_ENABLED
		workspace->root_uri = "file://" + workspace->root;
#else
		workspace->root_uri = "file:///" + workspace->root;
#endif
```
It assumes that `workspace_root` is always absolute, which seems like a reasonably safe assumption? It also assumes that the LSP client and server are on the same OS.

There is also the risk that this path format is actually different on different LSP client implementation on Windows.

Reply:
Alright, reading https://en.wikipedia.org/wiki/File_URI_scheme it does seem indeed that the prefix will often be `file:///` because it should be `file://` + `host` + `/` + `path` and `host` can be empty to default to localhost.

So `file:///c:/path/to/root` is indeed valid. And reading that page I would actually expect a Linux absolute path to be `file:////path/to/file` or `file://localhost//path/to/file`, but I guess they made the root slash optional.

So this fix looks good indeed.

Reply:
File URI format is defined in [RFC 8089](https://datatracker.ietf.org/doc/html/rfc8089) and according to [Wikipedia](https://en.wikipedia.org/wiki/File_URI_scheme) this is the correct way:
```
file:///c:/WINDOWS/clock.avi
```

Interestingly, the article also says:
> A valid file URI must therefore begin with either file:/path (no hostname), file:///path (empty hostname), or file://hostname/path.
> file://path (i.e. two slashes, without a hostname) is never correct, but is often used.